### DISCUSSION TITLE
Batch Montage Files Into a folder - please help

### DISCUSSION THREAD

**Discussion by JacobPrince1** [ImageMagick/ImageMagick#2090] [Help]
Trying to figure out how to work this properly. If someone could just write the command for what I'm trying to do that would be a great help.

This is what I conjured up:
`montage [filepaths] -density 600 -geometry +2+2 -path /Users/MyUsername/lyrics out.png`

I have 56 files in a folder that I want to montage into 14 2x2 png photos. The photos are roughly 600x600 pixels, I want the montage to be full quality with no reductions. The photos are alphabetically named in the order that I want them to display (i'm assuming they will go from left to right, like this:
1 | 2
3 | 4

first issue is `error unrecognized option -path`
I don't want the software to dump these photos in my main folder, I've never done batch edits like this before so in the past I manually move the output photos to my desired directory. How to I set the output folder correctly?

second issue is the filename
How do I batch name the files so they keep their order?

last issue
How do I add the input files without dragging and dropping them into terminal. That's the only way I can figure out to add a batch of photos, drag and drop. I think there is some way to specify the folder but I just can't seem to figure it out. 

Looking forward to your replies

**Comment by snibgo** [comment]
What version of IM, on what platform?

"-path" is an option for mogrify, but not montage. For montage, simply give your output file a path.

**Comment by fmw42** [comment]
You will have to write a script loop over all your image in the folder and take them 4 at a time. For each 4 images, use montage. The input image filenames you have to insert in montage from your script loop. Remove -path and just close up the path with the output name.

`montage image1 image2 image3 image4 -tile 2x2 -geometry +2+2  /Users/MyUsername/lyrics/out.png`

It looks like you are on Unix-like system when you use forward slashes. So you would need to use bash shell scripting for your "for" loop over all files in your folder.  The files would need to be in alphabetic order in the folder so that you could take groups of 4 for each loop. The output directory must be created before using ImageMagick. ImageMagick does not create output directories. The output name will need to be different for each group of 4 images.

Please identify your exact IM version and platform.
